subsys/bluetooth/host/l2cap.c

```diff
@@ -1295,9 +1295,11 @@ static void le_ecred_reconf_req(struct bt_l2cap *l2cap, uint8_t ident,
 		goto response;
 	}
 
-	while (chan_count-- >= 0) {
+	while (chan_count >= 1) {
+		chan_count -= 1;
 		BT_L2CAP_LE_CHAN(chans[chan_count])->tx.mtu = mtu;
 		BT_L2CAP_LE_CHAN(chans[chan_count])->tx.mps = mps;
+
 	}
```

Comment:
Why not use a for loop to make it more readable? 
```suggestion
	for (int i = 0; i < chan_count; i++) {
		BT_L2CAP_LE_CHAN(chans[i])->tx.mtu = mtu;
		BT_L2CAP_LE_CHAN(chans[i])->tx.mps = mps;
	}
```

Then the value of `chan_count` also remain valid.